MetaMask/metamask-extension
# Adding new icon components (WIP)

**TO DO:**
- [x] Rebase once `BaseIcon` has been merged https://github.com/MetaMask/metamask-extension/pull/15609
- [ ] Finalized icons. I have a list of missing or broken svgs that I need to update
- [ ] Add tests for generated icon components
- [ ] Finalized where the icon scripts will live

## Explanation

This PR adds new svg react components as apart of the IA/Nav efforts. It includes:

- `BaseIcon` component use as a base for all icon components
- icon svgs `ui/components/component-library/icons/svgs
- [SVGR](https://react-svgr.com/docs/getting-started/) package that generates react components from svg files
- scripts to generate the svg icon react components

## More Information

* Fixes #15098 
- [Figma icons file](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?node-id=401%3A2122)
## Screenshots/Screencaps

https://user-images.githubusercontent.com/8112138/184191046-680f2c16-5e39-47a8-a66e-7eeb834fe823.mov

## Manual Testing Steps

**To test building icons**
- Go to `ui/components/component-library/icons`
- Run `yarn build:icons` in the terminal
- See  `ui/components/component-library/icons/lib` has been created with svg react components

**To test storybook**
- Go to the storybook build on this PR
- Search for `icons`
- Select the first result
- Test icon search and code copy

## Pre-Merge Checklist


### + If there are functional changes:

- [ ] Manual testing complete & passed (not sure ??)
